sunpy/net/fido_factory.py

```diff
@@ -347,9 +347,9 @@ def fetch(self, *query_results, path=None, max_conn=5, progress=True,
         >>> filepaths = Fido.fetch(filepaths)  # doctest: +SKIP
 
         """
-        if path is not None and not os.access(path, os.W_OK):
+        if path is not None and not os.access(str(path).split("{")[0], os.W_OK):
             raise PermissionError('You do not have permission to write'
-                                  ' file in this directory')
+                                  f' to the directory {path}')
```

Comment:
I should probably change the error message too, so people aren't confused as to why the `{}` parts of the path are here but not checked for.